optaplanner-examples/src/main/java/org/optaplanner/examples/taskassigning_listvariable/score/TaskAssigningConstraintProvider.java

```diff
@@ -47,7 +47,7 @@
 
     private UniConstraintStream<Task> getTaskWithPriority(ConstraintFactory constraintFactory, Priority priority) {
         return constraintFactory.forEach(Task.class)
-                .filter(task -> task.getPriority() == priority);
+                .filter(task -> task.getEmployee() != null && task.getPriority() == priority);
```

Comment:
This is the only difference from 9ab908311b22acf121fcc340cedbc0c4c4f2b382.

I have to do this to avoid NPE in CH and because `forEach()` doesn't filter out shadow planning entities with uninitialized shadow variables.

@triceo Is this the most effective way to do that?

Reply:
Yeah, I do not see any other way.
That said, maybe it should be discussed? The point of `forEach(...)` is to eliminate null checks, and this seems like a corner case that escaped.

Reply:
According to Proposal I, this behavior was chosen on purpose. I have to say I don't fully understand the reasoning that explains Proposal I rejection but I assume it was understood during the discussion.

In other words, this corner case hasn't escaped our attention. Whether it is a design flaw or not, I'm not sure though.